### DISCUSSION TITLE
Can VitePress handle large projects?  (yes, building issues)

### DISCUSSION THREAD

**Discussion by luzdealba** [vuejs/vitepress#3189] [Q&A] [answered]
Hi!

My VitePress setting seems to be particularly large as I have tried everything and anything: more memory assignment with `--max-old-space-size`, `sourcemap` and other hacks in `vite.config.js`, and nothing seemed to work for me.

I'm well aware that this is a [Rollup](https://rollupjs.org/) issue and that there's lengthy discussion about this on #2433; nonetheless, Rollup is the builder VitePress uses, and it's not that clear that this is even on the radar for resolution.

So--is there an actual roadmap to address this issue in the forseable future, or--awesome as it is--shall I just assume that VitePress isn't designed for large projects ("large" basically being hundreds of Markdown files) and look elsewhere?

By the way, thanks to Evan, Kia, Divyansh and Eduardo for all the great work 🙏

**Comment by brc-dd** [comment, marked as answer]
It's definitely on our radar. But for v1 we are targeting stability more than performance. So, this would likely take some time to get fixed. I'm kind of hoping rolldown doesn't have this issue and when vite switches to it, vitepress automatically becomes more scalable. Otherwise we will have to come up with some sort of isolated or concurrent build strategy. If you have thousands of file, then yeah, for now you should use something else.

**Reply by luzdealba** [reply]
Hey Divyansh, thanks a lot for the clear and straightforward answer!

That clarifies it  :))
